fsspec/implementations/tests/data/diabetes.csv

```diff
@@ -0,0 +1,769 @@
+Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
+6,148,72,35,0,33.6,0.627,50,1
```

Comment:
Why include this big file? A small sample in code would be fine for the sake of testing.

Reply:
I was trying to replicate the error I was having in a real case scenario to stress the BDFS API writing concurrently many records, doing this triggered the 429 error, I could remove the file and only put a couple of records but it will not replicate the initial issue

Reply:
How about 
```
import concurrent.futures
fs = ...
ex = concurrent.futures.ThreadPoolExecutor()
ex.map(lambda i: fs.pipe(f"/path/{i}", b"data"), range(100))
```
You could use a mock to test that the retry is actually getting called; OR allow logging during retry (not a bad idea anyway) and check for logging statements coming out.

Reply:
Actually, we're using VCR, so you'll never get a response except the one you go the original time. I have no idea, then, how to get back a 429 without mocking